Gem/Code/Source/Components/NetworkAiComponent.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Source/Components/NetworkAiComponent.h>
+
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <Source/Components/WasdPlayerMovementComponent.h>
+#include <Source/Components/NetworkWeaponsComponent.h>
+#include <AzCore/Time/ITime.h>
+#include <Multiplayer/Components/NetBindComponent.h>
+
+namespace MultiplayerSample
+{
+    void NetworkAiComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<NetworkAiComponent, NetworkAiComponentBase>()->Version(1);
+        }
+
+        NetworkAiComponentBase::Reflect(context);
+    }
+
+    void NetworkAiComponent::OnInit()
+    {
+        // TODO: Provide an interface for controlling the seed
+        m_lcg.SetSeed(static_cast<uint64_t>(AZ::Interface<AZ::ITime>::Get()->GetElapsedTimeMs()));
+    }
+
+    void NetworkAiComponent::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    void NetworkAiComponent::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    void NetworkAiComponent::TickMovement(WasdPlayerMovementComponentController& movementController, float deltaTime)
+    {
+        // TODO: Execute this tick only if this component is owned by this endpoint (currently ticks on server only)
+        m_remainingTimeMs -= deltaTime * 1000;
+
+        if (m_remainingTimeMs <= 0)
+        {
+            // Determine a new directive after 500 to 9500 ms
+            m_remainingTimeMs = m_lcg.GetRandomFloat() * 9500.f + 500.f;
```

Comment:
those numbers would make good cvars.. we can hook cvars up to imgui sliders in menus as well

Reply:
I actually intended on hooking them up to ImGUI sliders but could not get the StressTestComponent to replicate to the client (which is where I would host that data). I can change them to cvars at the minimum before checkin.